app/models/miq_worker.rb

```diff
@@ -455,6 +455,17 @@ def clean_active_messages
     end
   end
 
+  def error_out_tasks_with_active_queue_message
+    message = "Task Handler: [#{friendly_name}] ID [#{id}] has been deleted!"
+    processed_messages.each do |m|
+      task = m.miq_task
+      next unless task
+
+      args = [MiqTask::STATE_FINISHED, MiqTask::STATUS_ERROR, message]
+      MiqQueue.put(:class_name => task.class.name, :instance_id => m.miq_task_id, :method_name => "update_status", :args => args)
```

Comment:
It feels a bit much to queue one for tasks here as `MiqTask#update_status` is pretty much a single update on the task row.  I guess I can queue up a list of tasks ids and do them all in one async queue message.

Or,  I can just inline call `update_status` directly.   🤷

Reply:
Yeah I'm not sure it's worth it to queue this up...I guess it depends on how many statuses you think there will be?

Reply:
It should be low normally as this is one worker destroy and a worker shouldn't be working on much more than one message at a time and likely, not many tasks.

Reply:
Note, I'll be looking at something similar on the miq_request and miq_request_task side but parsing the tracking_label to instantiate and update the resulting task/request so they should probably be done similarly.

Reply:
> and a worker shouldn't be working on much more than one message at a time and likely, not many tasks.

Queue workers can only work on one thing at a time, so it's literally only one queue item.  I assume that means only 1 task also.

Reply:
yeah, exactly... anything else is an exception